pallets/gear/src/internal.rs

```diff
@@ -796,7 +866,8 @@ where
 
         // If gas limit can cover threshold, message will be added to mailbox,
         // task created and funds reserved.
-        let expiration = if GasHandlerOf::<T>::exists(message.id()) {
+
+        let expiration = if to_mailbox {
```

Comment:
the reason why some gas still locked is because you don't consume node, once message sent to user not for mailbox within the function. Previously here wasn't node, now it is, so should be consumed!

Reply:
also write test for it

Reply:
this case is already covered with incorrect test. After fixing it covers correct case

Reply:
done